File: app/src/main/java/timothypaetz/com/recyclersectionheader/PersonAdapter.java
package timothypaetz.com.recyclersectionheader;

import android.support.annotation.LayoutRes;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import java.util.List;

/**
 * Created by paetztm on 2/6/2017.
 */
public class PersonAdapter extends RecyclerView.Adapter<PersonAdapter.ViewHolder> {

    private final List<Person>   people;
    private final LayoutInflater layoutInflater;
    private final int            rowLayout;

    public PersonAdapter(LayoutInflater layoutInflater, List<Person> people, @LayoutRes int rowLayout) {
        this.people = people;
        this.layoutInflater = layoutInflater;
        this.rowLayout = rowLayout;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View v = layoutInflater.inflate(rowLayout,
                                        parent,
                                        false);
        return new ViewHolder(v);
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        Person person = people.get(position);
        holder.fullName.setText(person.getFullName());
    }

    @Override
    public int getItemCount() {
        return people.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        private TextView fullName;

        public ViewHolder(View view) {
            super(view);
            fullName = (TextView) view.findViewById(R.id.full_name_tv);
        }
    }
}

File: app/src/main/java/timothypaetz/com/recyclersectionheader/PersonAdapter.kt
package timothypaetz.com.recyclersectionheader

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.annotation.LayoutRes
import androidx.recyclerview.widget.RecyclerView

/**
 * Created by paetztm on 2/6/2017.
 */
class PersonAdapter(
    private val layoutInflater: LayoutInflater,
    private val people: List<Person>,
    @param:LayoutRes private val rowLayout: Int
) : RecyclerView.Adapter<PersonAdapter.ViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val v = layoutInflater.inflate(
            rowLayout,
            parent,
            false
        )
        return ViewHolder(v)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val person = people[position]
        holder.fullName.text = person.fullName
    }

    override fun getItemCount(): Int {
        return people.size
    }

    class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val fullName: TextView = view.findViewById(R.id.full_name_tv)
    }
}